File: code_forces/Intersting_Drink.cpp
#include<iostream>
#include<algorithm>
#include<math.h>
#include<vector>
using namespace std;

int binary_search(vector<int>& prices,int target)
{
    int low =0 ,top=prices.size()-1,mid;
    while(low<=top)
    {
        mid=(low+top)/2;
        if(prices[mid]==target)
            return mid +1;
        else if (prices[mid]>target)
            top=mid-1;
        else
            low=mid+1;
    }
    return -1;
}

int main()
{
    int n; 
    cin >>n;
    vector<int> prices(n);
    for(int i =0;i<n;i++)
        cin >> prices[i];
    sort(prices.begin(),prices.end());
    int days;
    cin >> days;
    int budget;
    for(int i=0;i<days;i++){
        cin>>budget;
        cout<<binary_search(prices,budget)<<endl;
    }

}

File: code_forces/Intersting_Drink.py
import math


def binary_search(lis, target, target_1=None):
    low, top = 0, len(lis) - 1
    while low <= top:
        mid = (low + top) // 2
        if lis[mid] == target:
            if target_1 != None and target > target_1:
                return mid
            return mid + 1
        if lis[mid] < target:
            low = mid + 1
        if lis[mid] > target:
            top = mid - 1
    return binary_search(
        lis, lis[min(range(len(lis)), key=lambda i: abs(lis[i] - target))], target
    )


def main():
    trash_value = input()
    prices = list(map(int, input().split()))
    prices = sorted(prices)
    days = int(input())

    for i in range(days):
        budget = int(input())
        if budget < prices[0]:
            print(0)
            continue
        print(binary_search(prices, budget))


if __name__ == "__main__":
    main()
